### DISCUSSION TITLE
GOES rgb recipes doesn't work out of the box

### DISCUSSION THREAD

**Discussion by data-panda** [blaylockbk/goes2go#19] [Q&A] [answered]
Hi, I am hitting a roadblock where `goes_latest()` returned xarray object does not contain rgb as an attribute such that the call `G.rgb.TrueColor()` throws exception (or any methods within the `rgb.py`). Am I missing something very trivial which has changed recently? Because not being able to replicate the very first few lines of the RGB demo is a bummer.

**Comment by blaylockbk** [comment, marked as answer]
Hi @data-panda, sorry you are hitting a roadblock. 

I'm not able to reproduce your problem. This is what I did...

```python
from goes2go.data import goes_latest
import matplotlib.pyplot as plt
```
```python
ds = goes_latest()
```
![image](https://user-images.githubusercontent.com/6249613/165840034-3cb12e00-687e-424c-bc52-39b9c0df3020.png)
```python
ds.rgb
```
![image](https://user-images.githubusercontent.com/6249613/165839650-c6029978-4ab6-4bab-babc-95598ec7125e.png)

```python
rgb = ds.rgb.TrueColor()
rgb
```
![image](https://user-images.githubusercontent.com/6249613/165840073-cf4c0619-1f3f-4ef8-8aea-e22779356eb4.png)

```python
plt.imshow(rgb)
```
![image](https://user-images.githubusercontent.com/6249613/165840104-bdd3e786-d265-497b-b43a-27be1d08a8e0.png)


---
That said, the `rgb` attribute in the statement `ds.rgb.TrueColor()` is an xarray Dataset accessor, which should be loaded when you import goes2go.

These lines are at the top of __init__.py when you import goes2go
```
try:
    import goes2go.accessors
except:
    warnings.warn("goes2go xarray accessors could not be imported.")
```

If you are getting the warning that **goes2go xarray accessors could not be imported**, then that would be your problem. 

These are imported by the accessor; if any of these are not in your python environment, that could cause the problem
```python
import warnings
import cartopy.crs as ccrs
import numpy as np
import xarray as xr
from shapely.geometry import Point, Polygon
```

**Reply by data-panda** [reply]
Indeed the issue was with cartopy not being in the environment.
